projects/RabbitMQ.Client/client/impl/ContentHeaderPropertyReader.cs

```diff
@@ -78,82 +80,81 @@ public bool ReadBit()
             return ReadPresence();
         }
 
-        public void ReadFlagWord()
+        private void ReadBits()
         {
             if (!ContinuationBitSet)
             {
                 throw new MalformedFrameException("Attempted to read flag word when none advertised");
             }
-            m_flagWord = NetworkOrderDeserializer.ReadUInt16(_memory.Slice(_memoryOffset).Span);
-            _memoryOffset += 2;
-            m_bitCount = 0;
+            _bits = NetworkOrderDeserializer.ReadUInt16(Span);
+            _offset += 2;
+            _bitMask = StartBitMask;
         }
 
         public uint ReadLong()
         {
-            uint result = NetworkOrderDeserializer.ReadUInt32(_memory.Slice(_memoryOffset).Span);
-            _memoryOffset += 4;
+            uint result = NetworkOrderDeserializer.ReadUInt32(Span);
+            _offset += 4;
             return result;
         }
 
         public ulong ReadLonglong()
         {
-            ulong result = NetworkOrderDeserializer.ReadUInt64(_memory.Slice(_memoryOffset).Span);
-            _memoryOffset += 8;
+            ulong result = NetworkOrderDeserializer.ReadUInt64(Span);
+            _offset += 8;
             return result;
         }
 
         public byte[] ReadLongstr()
         {
-            byte[] result = WireFormatting.ReadLongstr(_memory.Slice(_memoryOffset));
-            _memoryOffset += 4 + result.Length;
+            byte[] result = WireFormatting.ReadLongstr(Span);
```

Comment:
Seems to still alloc `byte[]` does it make sense to remove the `ToArray()` underneath or not worth it because the method is not used anyway? See my other comment

Reply:
If it is not used, I rather delete it than change it. But in general we could return a Memory<Byte>, but also here, it's a different contract we implicitly get by doing so.

Reply:
Yeah but this is a purely internal thing . Maybe let's wait until Michael and Luke chime in

Reply:
IMHO change internal contracts as you see fit :+1:

Reply:
Sounds like a fairly minor implementation detail to me? I trust @bollhals' judgment on this then :)